src/ucp/api/ucp_def.h

```diff
@@ -13,17 +15,124 @@
 
 #define UCP_REQUEST_PRIV_LEN             80
 
-
-typedef struct ucp_params                ucp_params_t;
+/**
+ * @ingroup UCP_CONTEXT
+ * @brief UCP Application Context
+ *
+ * UCP application context (or just a context) is an opaque handle that holds a
+ * UCP communication instance’s global information.  It represents a single UCP
+ * communication instance.  The communication instance could be an OS process
+ * (an application) that uses UCP library.  This global information includes
+ * communication resources, endpoints, memory, temporary file storage, and
+ * other communication information directly associated with a specific UCP
+ * instance.  The context also acts as an isolation mechanism, allowing
```

Comment:
We can add here that this allows for multiple independent communications capability